src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,157 @@
+        return getNodeByIndex(index).item;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        Node<T> current = getNodeByIndex(index);
+        T oldObject = current.item;
+        current.item = value;
+        return oldObject;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        Node<T> current = getNodeByIndex(index);
+        if (current.next == null && current.prev == null) {
+            head = null;
+            tail = null;
+        } else if (current.next == null) {
+            current.prev.next = null;
+            tail = current.prev;
+        } else if (current.prev == null) {
+            current.next.prev = null;
+            head = current.next;
+        } else {
+            current.prev.next = current.next;
+            current.next.prev = current.prev;
+        }
+        size--;
+        return current.item;
     }
 
     @Override
     public boolean remove(T object) {
+        Node<T> current = head;
+        for (int i = 0; i < size; i++) {
+            if ((object == current.item)
+                    || (object != null && object.equals(current.item))) {
+                remove(i);
+                return true;
+            }
+            current = current.next;
+        }
         return false;
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return size == EMPTY_LIST_VALUE;
+    }
+
+    private static class Node<T> {
+        private T item;
+        private MyLinkedList.Node<T> next;
+        private MyLinkedList.Node<T> prev;
+
+        Node(MyLinkedList.Node<T> prev, T element, MyLinkedList.Node<T> next) {
+            this.item = element;
+            this.next = next;
+            this.prev = prev;
+        }
+    }
+
+    private Node<T> getNodeByIndex(int index) {
+        checkIndexExistence(index);
+        boolean isIndexLessHalf = index <= size / DEFAULT_DIVISOR;
+        Node<T> node;
+        if (isIndexLessHalf) {
+            node = head;
+            while (index != EMPTY_LIST_VALUE) {
+                node = node.next;
+                index--;
+            }
+        } else {
+            node = tail;
+            int stepCounter = size - index;
+            while (stepCounter != INDEX_STOPPING_FROM_TAIL) {
+                node = node.prev;
+                stepCounter--;
+            }
+        }
+        return node;
+    }
+
+    private void checkAddIndexExistence(int index) {
+        if (index < 0 || index > size) {
+            throw new IndexOutOfBoundsException(String
+                    .format(INDEX_OUT_OF_BOUNDS_MESSAGE, index));
+        }
+    }
+
+    private void checkIndexExistence(int index) {
+        if (index < 0 || index >= size) {
+            throw new IndexOutOfBoundsException(String
+                    .format(INDEX_OUT_OF_BOUNDS_MESSAGE, index));
+        }
```

Comment:
You can connect these two methods in one. Think what you should give as parameters when you invoke this method.